### PR TITLE
[MRG+1] default return value for extract_first

### PR DESCRIPTION
**Pull request by bosnj**
It would be useful to have a default return value for `extract_first`, to avoid additional data type checks. 

For example, when expecting a string, this would raise an exception:
`something = selector.extract_first().lower()`

Allowing an optional `default` value can solve this elegantly:
`something = selector.extract_first(default='').lower()`

### CODE REVIEW

**Comment by pablohoffman** [comment]
I like it.

**Comment by nramirezuy** [comment]
@bosnj Simple test plz :+1:

**Comment by Curita** [comment]
I like it too. Should we add a default to `re_first` as well?

Documenting it would be good too, a simple use-case (like the one on this issue's description) in `docs/topics/selectors.rst` after introducing .extract_first() should be enough.

**Comment by kmike** [comment]
I'm fine with this feature. 

1. It needs tests.
2. I think that we should use  keyword arguments in examples and documentation . 

`selector.extract_first('').lower()` doesn't look good to me, but `selector.extract_first(default='').lower()` does.

**Comment by kmike** [comment]
Also, it may be a right time to start adding docstrings to new methods/features.

File: scrapy/selector/unified.py [hunk 1]
```diff
@@ -182,9 +182,9 @@ def re_first(self, regex):
     def extract(self):
         return [x.extract() for x in self]
 
-    def extract_first(self):
+    def extract_first(self, default=None):
         for x in self:
-            return x.extract()
+            return x.extract() or default
```
Comment (dangra) [inline scrapy/selector/unified.py]: It doesn't look good to me. It is returning the default value only if first matched selection is an empty string.

I think the default value should be used only when there aren't matches like in:

```python
for x in self:
    return x.extract()
else:
    return default
```

**Comment by bosnj** [comment]
Thanks @dangra, updated as per your suggestion.
Also added a test case and a mention in the docs.

**Comment by Curita** [comment]
LGTM. I'd like the same feature in `re_first` but we can implement it in another pull request.

**Comment by kmike** [comment]
let's sneak this in :)

**Comment by dangra** [comment]
:beetle:juice x 3

**Pull request merged**